### PR TITLE
fix: Fix and re-enable previously broken tests

### LINKED ISSUE #1072: Firebase-backend branch: Re-enable unit tests related to ready state detection
https://github.com/googleapis/cloud-debug-nodejs/pull/1067 disabled some unit tests related to ready state detection.  The refactors of the controller code have broken that functionality and must be repaired before the Firebase-backend branch can be merged into the main branch.

This is required as a part of developing https://github.com/googleapis/cloud-debug-nodejs/issues/1071

### PR DESCRIPTION
**Pull request by mctavish**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/cloud-debug-nodejs/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #1072 🦕

### CODE REVIEW

File: src/agent/debuglet.ts [hunk 1]
```diff
@@ -765,8 +765,8 @@ export class Debuglet extends EventEmitter {
             }
           ).debuggee.id;
           // TODO: Handle the case when `result` is undefined.
-          that.emit('registered', (result as {debuggee: Debuggee}).debuggee.id); // FIXME: Do we need this?
-          that.debuggeeRegistered.resolve(); // FIXME: Do we need this?
+          that.emit('registered', (result as {debuggee: Debuggee}).debuggee.id);
```
Comment (jasonborg) [inline src/agent/debuglet.ts:768]: It may make sense to add a comment here for either or both lines

File: src/agent/debuglet.ts [hunk 2]
```diff
@@ -788,9 +788,15 @@ export class Debuglet extends EventEmitter {
             err.name === 'RegistrationExpiredError'
               ? 0
               : this.config.internal.registerDelayOnFetcherErrorSec;
+          this.updatePromise();
```
Comment (jasonborg) [inline src/agent/debuglet.ts:791]: Consider adding a comment for this,

**Review by jasonborg — APPROVED** [review]

File: src/agent/debuglet.ts (hunk 1 above)
Comment (mctavish) [reply src/agent/debuglet.ts:768]: It's non-obvious at this code location, but the events are documented at the constructor and the promises are documented where they are declared.  I've added a comment at with the promises to help provide context for why they exist.

File: src/agent/debuglet.ts (hunk 2 above)
Comment (mctavish) [reply src/agent/debuglet.ts:791]: Done.

**Pull request merged**